site/sfguides/src/getting_started_with_traces/getting_started_with_traces.md

```diff
@@ -0,0 +1,443 @@
+## Run the Procedure
+Duration: 2
+
+Create a new python cell at the bottom of the notebook by clicking on + Python.
+
+```python
+from snowflake.snowpark.context import get_active_session
+
+session = get_active_session()
+session.sql("call get_weather_for_cities_sp('bronze_weather_api', 1, 15)").collect()
+session.table("bronze_weather_api")
+
+```
+
+Verify it is successful. After running, the notebook will output the dataframe of bronze_weather_api which includes the current weather for all the cities in city_list.
+
+<!-- ------------------------ -->
+## Query the Data Ingested
+Duration: 2
+
+Create a new sql cell by clicking on + SQL.
+
+Paste this section into the new cell.
+
+```sql
+SELECT NAME, 
+       CURRENT_WEATHER['main']['temp']::float as KELVIN_TEMP, 
+       CURRENT_WEATHER['weather'][0]['main']::varchar as CONDITIONS 
+FROM bronze_weather_api
+QUALIFY ROW_NUMBER() over (partition by NAME order by CURRENT_WEATHER['dt'] desc) = 1;
+
+```
+
+Run this cell to verify it is successful.
+
+This will show the latest weather pulled by the notebook.
+
+
+<!-- ------------------------ -->
+## View the Traces
+Duration: 5
+
+Click on the left navigation item "Monitoring" > "Traces & Logs".
+
+The top entry in the traces should be the calls from the Current Notebook, named `TRACING_QUICKSTART`. Traces can take a minute to show up, so you may have to wait and refresh until it appears. You may have to filter using the `Database` filter and selecting `TRACING_QUICKSTART` if there are other traces in the account.
+
+Click on Trace for the Notebook. Select the Span Type Filters at the top to include UDF and Procedures. Excluding Streamlit will remove the notebook calls from the view. Look for the `get_weather_for_cities_sp` span and the `__main__:get_weather_for_all_cities` spans which are the more time consuming parts of the procedure. You can see fetching the data from the API and saving is only a few seconds, but the entire runtime of the procedure is over a minute.
+
+If you expand the `get_weather_fn` entry under `__main__:get_weather_for_all_cities`, you can see the Debug Logs by clicking on Logs on the trace to see all the http request information.
+
+This tracing information shows the entire execution timeline with information on every call.
+
+<img src="assets/trace_details.png" width="889" height="513">
+
+<!-- ------------------------ -->
+## Query the Metrics, Logs, and Trace Data
+Duration: 3
+
+Memory and CPU metrics are also available because the session set the METRIC_LEVEL to all.
+
+To get to this data, it is in the Events table.
+
+Find your events table by running this sql:
+
+```sql
+SHOW PARAMETERS LIKE 'event_table' IN ACCOUNT;
+
+```
+
+Query the memory and cpu metrics with this query, replacing `YOUR_EVENT_TABLE` with your event table found in the previous query:
+
+```sql
+SELECT *
+FROM YOUR_EVENT_TABLE
+WHERE
+    RECORD_TYPE = 'METRIC' 
+    and RESOURCE_ATTRIBUTES['db.user'] = '{{user_name}}' 
```

Comment:
Should we add a filter for query id
`resource_attributes['snow.query.id'] = {{query_id}}`
or function / procedure name
`resource_attributes['snow.executable.name'])::string = {{executable_name}}` as well?

Reply:
with the database filters, i can't imagine any other parallel work needing filtered